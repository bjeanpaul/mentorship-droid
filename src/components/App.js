import React from 'react-native';
import { Provider } from 'react-redux';

import configureStore from '../stores/configureStore';
const store = configureStore();


console.log('----------------------------')
import { login } from '../mentor/MentorActions';
store.dispatch(login('admin', '123'));
console.log('----------------------------')


import RootNavigator from './RootNavigator';



const App = function App() {
  return (
    <Provider store={store}>
        <RootNavigator />
    </Provider>
  );
};

export default App;
