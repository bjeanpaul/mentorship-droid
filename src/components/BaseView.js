import React from 'react';
import { View, StyleSheet, StatusBar } from 'react-native';
import colors from 'src/constants/colors';


const styles = StyleSheet.create({
  view: {
    backgroundColor: colors.DEFAULT_BG,
    flex: 1,
  },
});

const BaseView = ({
  children,
  style,
  statusBarBackgroundColor = colors.DEFAULT_STATUS_BG,
}) => (
  <View style={[styles.view, style]}>
    { statusBarBackgroundColor
      ? <StatusBar backgroundColor={statusBarBackgroundColor} />
      : null
    }
    {children}
  </View>
);
BaseView.propTypes = {
  statusBarBackgroundColor: React.PropTypes.string,
  children: React.PropTypes.any,
  style: React.PropTypes.any,
};
export default BaseView;
