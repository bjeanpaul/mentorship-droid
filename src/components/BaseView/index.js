import React from 'react';
import { View } from 'react-native';

import styles from './styles';


const BaseView = ({
  children,
  style,
}) => (
  <View style={[styles.view, style]}>
    {children}
  </View>
);


BaseView.propTypes = {
  children: React.PropTypes.any,
  style: React.PropTypes.any,
};


export default BaseView;
