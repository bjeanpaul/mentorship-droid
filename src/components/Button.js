import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View, StyleSheet } from 'react-native';

import Text from 'Text';
import { font, fontWeight } from 'src/app/styles';


const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f77040',
    borderRadius: 5,
    marginLeft: 16,
    marginRight: 16,
  },
  buttonDisabled: {
    backgroundColor: '#dfe5e6',
  },
  label: {
    color: '#fff',
    fontFamily: font.brandonTextBold,
    fontSize: 13,
    fontWeight: fontWeight.bold,
    letterSpacing: 1,
    marginTop: 15,
    marginBottom: 17,
  },
});

const Button = ({
  label,
  disabled,
  handlePress,
}) => {
  let button = (
    <View style={[styles.button, disabled && styles.buttonDisabled]}>
      <Text style={styles.label}>{label.toUpperCase()}</Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={handlePress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};
Button.propTypes = {
  label: PropTypes.string.isRequired,
  handlePress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};
export default Button;
