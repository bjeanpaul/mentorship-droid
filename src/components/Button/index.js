import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View } from 'react-native';
import Text from 'src/components/Text';
import styles, { themes, layouts } from './styles';


const Button = ({
  theme = themes.default,
  layout = layouts.default,
  children,
  disabled,
  onPress,
}) => {
  let button = (
    <View
      style={[
        styles.container,
        theme.container,
        layout,
        disabled && theme.containerIsDisabled,
      ]}
    >
      <Text
        style={[
          styles.title,
          theme.title,
          disabled && theme.titleIsDisabled,
        ]}
      >
        {children}
      </Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={onPress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};


Button.themes = themes;
Button.layouts = layouts;


Button.propTypes = {
  children: PropTypes.any,
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  theme: PropTypes.any,
  layout: PropTypes.any,
};


export default Button;
