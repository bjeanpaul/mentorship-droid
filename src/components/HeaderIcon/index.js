import React, { PropTypes } from 'react';
import { View, Image } from 'react-native';

import Icon from 'src/components/Icon';
import styles from './styles';

import backLight from 'app/assets/back-light.png';


const HeaderIcon = ({
  type,
  style = {}
}) => (
  <Image source={type} style={[styles.icon, style]} />
)


HeaderIcon.types = Icon.types;


HeaderIcon.propTypes = {
  type: PropTypes.number.isRequired,
  style: PropTypes.object,
};


export default HeaderIcon;
