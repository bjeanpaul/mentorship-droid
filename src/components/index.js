import Text from './Text';
import Label from './Label';
import Link from './Link';
import Button from './Button';
import TextInput from './TextInput';
import MultiLineTextInput from './MultiLineTextInput';
import BaseView from './BaseView';
import Toolbar from './Toolbar';
import Header from './Header';
import Title from './Title';
import Icon from './Icon';
import HeaderIcon from './HeaderIcon';
import OverlayCompleted from './Overlay/Completed';
import OverlayLoading from './Overlay/Loading';
import Section from './Section';

export {
  BaseView,
  Toolbar,
  Text,
  Label,
  Link,
  Button,
  TextInput,
  MultiLineTextInput,
  Toolbar,
  Header,
  Title,
  Icon,
  HeaderIcon,
  OverlayCompleted,
  OverlayLoading,
  Section,
};
