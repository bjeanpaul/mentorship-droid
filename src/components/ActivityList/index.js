import React, { PropTypes } from 'react';
import { View, Image, ScrollView, TouchableNativeFeedback } from 'react-native';
import { Text } from 'src/components';
import images from 'src/constants/images';
import { ImageUrl } from 'src/api';
import styles from './styles';


const ActivityList = ({
  category: { color },
  activities,
  onActivityPress,
}) => (
  <ScrollView>
    {activities.map(({ id, title, icon, isComplete }) => (
    <TouchableNativeFeedback
      key={id}
      uid={`activity:${id}`}
      onPress={() => onActivityPress(id)}
    >
      <View style={styles.activity}>
        <ActivityIcon icon={icon} color={color} isComplete={isComplete} />
        <Text numberOfLines={3} style={styles.activityTitle}>{title}</Text>
      </View>
    </TouchableNativeFeedback>
    ))}
  </ScrollView>
);


const ActivityIcon = ({
  icon,
  color,
  isComplete,
}) => (
  <View style={styles.activityIconContainer}>
    <View
      style={[
        styles.activityIcon,
        isComplete
          ? { backgroundColor: color }
          : styles.activityIconIsIncomplete,
      ]}
    >
      <Image
        source={icon.toSource(images.ACTIVITY_ICON_FALLBACK)}
        style={styles.activityIconImage}
      />
    </View>

    {
      isComplete
        ? <Image
          style={styles.activityCompleteAnnotation}
          source={images.ACTIVITY_COMPLETE_ANNOTATION}
        />
        : null
    }
  </View>
);


ActivityList.propTypes = {
  category: PropTypes.object.isRequired,
  activities: PropTypes.array.isRequired,
  onActivityPress: PropTypes.func.isRequired,
};


ActivityIcon.propTypes = {
  icon: PropTypes.instanceOf(ImageUrl),
  color: PropTypes.string.isRequired,
  isComplete: PropTypes.bool.isRequired,
};


export default ActivityList;
