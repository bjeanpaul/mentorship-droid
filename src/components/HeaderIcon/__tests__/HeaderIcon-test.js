import React from 'react';
import { Header, HeaderIcon } from 'src/components';


describe('HeaderIcon', () => {
  it('should render', () => {
    expect(render(
      <Header theme={Header.themes.dark}>
        <HeaderIcon type={HeaderIcon.types.backLight} />
      </Header>
    )).toMatchSnapshot();
  });
});
