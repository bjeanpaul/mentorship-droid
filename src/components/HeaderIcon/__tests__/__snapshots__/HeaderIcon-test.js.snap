exports[`HeaderIcon should render 1`] = `
<View
  style={
    Array [
      Object {
        "height": 72,
        "justifyContent": "center",
        "paddingLeft": 16,
        "paddingRight": 16
      },
      Object {
        "backgroundColor": "#003035"
      },
      Object {}
    ]
  }>
  <Image
    source={1}
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 24
        },
        Object {}
      ]
    } />
</View>
`;
